### PR TITLE
rpm: Packaging fixes

### PR DESCRIPTION
**Pull request by branto1**
We should package (/var/)run/ceph and we should not pass interpret to postun scripts, see the commit messages for the details.

I'm currently waiting for gitbuilders to build this so that I could test the packages.

### CODE REVIEW

**Comment by branto1** [comment]
The first patch does not fix the denial for me, yet -> marking DNM.

**Comment by branto1** [comment]
I tested the latest build and it fixed the issues I've been seeing.

**Comment by branto1** [comment]
@ktdreyer @smithfarm care to review?

**Comment by smithfarm** [comment]
The first commit reverts a53d932fc4a77a1fac33c4f650f2bf629a2f1234 . . . I have the same commit downstream right now. (It doesn't work the way it's supposed to because the so name doesn't match or something.)

Anyway I compared your commit with mine manually and they look the same, but it would be nice if you redid the commit using "git revert" so the commit message starts like this:

```
Revert "rpm: put /sbin/ldconfig into -p"
    
This reverts commit a53d932fc4a77a1fac33c4f650f2bf629a2f1234.
```

To that you could append your description from 194fd05.

**Comment by ktdreyer** [comment]
Good catch @smithfarm re: that commit message.

Regarding "restore the context manually" (for the `/run` dir), does that mean the context is going to get wiped again the next time the box reboots?

**Comment by branto1** [comment]
@smithfarm It does not work the way it is supposed to because there are # before the section starts and those get sent as an input to ldconfig, I'll amend the commit message to include the Revert bits.

@ktdreyer Nope, there is no problem with that, there. The policy will be active after the reboot and the directory gets labelled properly. You need to relabel in packaging as those files don't get labelled properly straight away -- the policy is not active when they are being installed, yet.

**Comment by ktdreyer** [comment]
Got it, thanks!

`Reviewed-by: Ken Dreyer <kdreyer@redhat.com>`

File: ceph.spec.in [hunk 1]
```diff
@@ -1222,9 +1222,11 @@ fi
...
 
-%post -n librados2 -p /sbin/ldconfig
+%post -n librados2
+/sbin/ldconfig
 
-%postun -n librados2 -p /sbin/ldconfig
+%postun -n librados2
+/sbin/ldconfig
 
 #################################################################################
```
Comment (smithfarm) [inline ceph.spec.in]: @branto1 You mean these `#` ?

**Comment by smithfarm** [comment]
@branto1 My theory was that when ldconfig complains "relative path `0' used to build cache" it's referring to the numeric parameter that RPM passes to the %postun script.

**Comment by smithfarm** [comment]
When I do `/sbin/ldconfig ######################################` the error is different:

```
/sbin/ldconfig: Can't create temporary cache file /etc/ld.so.cache~: Permission denied
```

Anyway, the time savings is not worth the trouble IMO, so I agree that reverting is the best solution.

**Comment by smithfarm** [comment]
LGTM

**Comment by branto1** [comment]
@smithfarm see e.g. https://bugzilla.redhat.com/show_bug.cgi?id=1003962#c0

Basically, when you use non-default interpret and put something (asides from whitespace) in the body of the script then 0/1 is passed to the new interpret.

**Comment by yuriw** [comment]
some testes are blocked and waiting for this PR merge, pls merge if ready

**Pull request merged**